### PR TITLE
pipx/pipx_info: multiple fixes

### LINKED ISSUE #9009: pipx: Injecting packages fails when using global
### Summary

I took the integration tests from https://github.com/russoz-ansible/community.general/blob/main/tests/integration/targets/pipx/tasks/testcase-injectpkg.yml and try using them with `global: true` but they fail.

This is probably related to the other in which ansible is creating the command to execute with `pipx`. 

The following commands are *not* the same:

```console
pipx list --global
pipx --global list
```

The correct one is the first one, when the playbooks runs it's invoking the 2nd one. Another note, the integration tests from this collection are missing a test for testing for injected global packages.

This is a known issue reported in the following issue:
- https://github.com/pypa/pipx/issues/1443

### Issue Type

Bug Report

### Component Name

pipx

### Ansible Version

```console
ansible [core 2.17.5]
  config file = None
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible
  ansible collection location = /home/ubuntu/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = True
```

### Community.general Version

```console
ansible-galaxy collection list community.general

# /usr/local/lib/python3.10/dist-packages/ansible_collections
Collection        Version
----------------- -------
community.general 9.4.0 
```

### Configuration

_No response_

### OS / Environment

Ubuntu 22.04

### Steps to Reproduce

```yaml
---
- name: Test global inject
  hosts: localhost
  gather_facts: false
  become: true
  tasks:
  - name: ensure application pylint is uninstalled
    community.general.pipx:
      name: pylint
      state: absent
      global: true

  - name: install application pylint
    community.general.pipx:
      name: pylint
      global: true
    register: install_pylint

  - name: inject packages
    community.general.pipx:
      state: inject
      name: pylint
      global: true
      inject_packages:
        - licenses
```

### Expected Results

Expected licenses to be installed in the global venv of pylint.

### Actual Results

```console
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Test global inject] *********************************************************************************************************************************************************

TASK [ensure application pylint is uninstalled] ***********************************************************************************************************************************
ok: [localhost]

TASK [install application pylint] *************************************************************************************************************************************************
changed: [localhost]

TASK [inject packages] ************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.community.general.plugins.module_utils.mh.exceptions.ModuleHelperException
fatal: [localhost]: FAILED! => {"application": {}, "changed": false, "force": false, "index_url": null, "msg": "Trying to inject packages into a non-existent application: pylint", "name": "pylint", "output": {"application": {}, "force": false, "index_url": null, "name": "pylint", "source": null}, "source": null, "vars": {"application": {}, "force": false, "index_url": null, "name": "pylint", "source": null}}

PLAY RECAP ************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

ubuntu@ubuntu:~/Desktop/git/pipxtest$ pipx list --global
venvs are in /opt/pipx/venvs
apps are exposed on your $PATH at /usr/local/bin
manual pages are exposed at /usr/local/share/man
   package pylint 3.3.1, installed using Python 3.10.12
    - pylint (symlink missing or pointing to unexpected location)
    - pylint-config (symlink missing or pointing to unexpected location)
    - pyreverse (symlink missing or pointing to unexpected location)
    - symilar (symlink missing or pointing to unexpected location)
ubuntu@ubuntu:~/Desktop/git/pipxtest$ pipx --global list
nothing has been installed with pipx 😴
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

The integration tests were flawed, not really testing the latest features, specially `global`.

Fixes #9009 

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Refactoring Pull Request
- Test Pull Request

##### COMPONENT NAME

pipx
pipx_info

### CODE REVIEW

File: plugins/modules/pipx_info.py [hunk 1]
```diff
@@ -98,6 +98,14 @@
...
       sample:
         licenses: "0.6.1"
+    pinned:
+      description:
+      - Whether the installed application is pinned or not.
+      - When using C(pipx<=1.6.0), this returns C(null).
+      returned: success
+      type: bool
+      sample:
+        pinned: true
```
Comment (felixfontein) [inline plugins/modules/pipx_info.py:108]:
```suggestion
        pinned: true
      version_added: 10.0.0
```

File: tests/integration/targets/pipx/tasks/main.yml [hunk 2]
```diff
@@ -3,10 +3,28 @@
...
-    name: pipx
-    extra_args: --user
+- name: Determine pipx level
+  block:
+    - name: Install pipx>=1.7.0
+      pip:
+        name: pipx>=1.7.0
+        extra_args: --user
+    - name: Ensure path
+      ansible.builtin.command: "{{ ansible_python_interpreter }} -m pipx ensurepath"
```
Comment (felixfontein) [inline tests/integration/targets/pipx/tasks/main.yml:13]: This seems to cause problems on FreeBSD; it seems to hang there.

Comment (russoz) [reply tests/integration/targets/pipx/tasks/main.yml:13]: Yup, I noticed. I am trying to get my FreeBSD VMs running in vagrant again to test this locally, bumping into C compiler issues.

Comment (felixfontein) [reply tests/integration/targets/pipx/tasks/main.yml:13]: Good luck :)

File: plugins/modules/pipx_info.py [hunk 3]
```diff
@@ -98,6 +98,15 @@
       type: dict
       sample:
         licenses: "0.6.1"
+    pinned:
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/pipx_info.py:101]: Return value (not option)

[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/pipx_info.py:101]: Right, sorry :) It's still not mentioned ;)

File: changelogs/fragments/9044-pipx-fixes.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+minor_changes:
+  - pipx - refactor out parsing of ``pipx list`` output to module utils (https://github.com/ansible-collections/community.general/pull/9044).
+  - pipx_info - refactor out parsing of ``pipx list`` output to module utils (https://github.com/ansible-collections/community.general/pull/9044).
```
Comment (russoz) [inline changelogs/fragments/9044-pipx-fixes.yml:3]:
```suggestion
  - pipx_info - refactor out parsing of ``pipx list`` output to module utils (https://github.com/ansible-collections/community.general/pull/9044).
  - pipx_info - add new return value``pinned`` (https://github.com/ansible-collections/community.general/pull/9044).
```

File: plugins/modules/pipx_info.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/pipx_info.py:101]: Done

File: changelogs/fragments/9044-pipx-fixes.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  - pipx_info - add new return value``pinned`` (https://github.com/ansible-collections/community.general/pull/9044).
```
Comment (felixfontein) [inline changelogs/fragments/9044-pipx-fixes.yml:4]:
```suggestion
  - pipx_info - add new return value ``pinned`` (https://github.com/ansible-collections/community.general/pull/9044).
```

File: changelogs/fragments/9044-pipx-fixes.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+bugfixes:
+  - pipx module utils - add missing command line formatter for arguement ``spec_metadata`` (https://github.com/ansible-collections/community.general/pull/9044).
```
Comment (felixfontein) [inline changelogs/fragments/9044-pipx-fixes.yml:6]:
```suggestion
  - pipx module utils - add missing command line formatter for argument ``spec_metadata`` (https://github.com/ansible-collections/community.general/pull/9044).
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**